patches/minecraft/net/minecraft/world/entity/npc/VillagerTrades.java.patch

```diff
@@ -0,0 +1,18 @@
+--- a/net/minecraft/world/entity/npc/VillagerTrades.java
++++ b/net/minecraft/world/entity/npc/VillagerTrades.java
+@@ -1367,6 +_,7 @@
+         private final int villagerXp;
+ 
+         public EmeraldsForVillagerTypeItem(int p_35669_, int p_35670_, int p_35671_, Map<VillagerType, Item> p_35672_) {
++            if (false) // FORGE: Modders can add custom villager types, so remove this validation
+             BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_327046_ -> {
+                 throw new IllegalStateException("Missing trade for villager type: " + BuiltInRegistries.VILLAGER_TYPE.getKey(p_327046_));
+             });
+@@ -1381,6 +_,7 @@
+         public MerchantOffer getOffer(Entity p_219685_, RandomSource p_219686_) {
+             if (p_219685_ instanceof VillagerDataHolder villagerdataholder) {
+                 ItemCost itemcost = new ItemCost(this.trades.get(villagerdataholder.getVillagerData().getType()), this.cost);
++                if (itemcost.itemStack().isEmpty()) return null; // FORGE: Account for modded villager types by returning null if there is no trade
```

Comment:
So, this will not work. 
get returns null
ItemConst init calls `p_332783_.asItem().builtInRegistryHolder()` which will NPE.
So this null check need to be brought up before the ItemCost init.
Larger patch but something like
```diff
-ItemCost itemcost = new ItemCost(this.trades.get(villagerdataholder.getVillagerData().getType()), this.cost);
+var item = this.trades.get(villagerdataholder.getVillagerData().getType());
+if (item == null) return null;
+ItemCost itemcost = new ItemCost(item, this.cost);
```